[2023-04-28T00:43:32.925+0000] {taskinstance.py:1090} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark-example.sql_job scheduled__2023-04-27T00:00:00+00:00 [queued]>
[2023-04-28T00:43:32.933+0000] {taskinstance.py:1090} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark-example.sql_job scheduled__2023-04-27T00:00:00+00:00 [queued]>
[2023-04-28T00:43:32.933+0000] {taskinstance.py:1288} INFO - 
--------------------------------------------------------------------------------
[2023-04-28T00:43:32.933+0000] {taskinstance.py:1289} INFO - Starting attempt 1 of 1
[2023-04-28T00:43:32.934+0000] {taskinstance.py:1290} INFO - 
--------------------------------------------------------------------------------
[2023-04-28T00:43:32.948+0000] {taskinstance.py:1309} INFO - Executing <Task(SparkSqlOperator): sql_job> on 2023-04-27 00:00:00+00:00
[2023-04-28T00:43:32.954+0000] {standard_task_runner.py:55} INFO - Started process 103 to run task
[2023-04-28T00:43:32.958+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'spark-example', 'sql_job', 'scheduled__2023-04-27T00:00:00+00:00', '--job-id', '28', '--raw', '--subdir', 'DAGS_FOLDER/spark-example.py', '--cfg-path', '/tmp/tmpr6nog3p6']
[2023-04-28T00:43:32.960+0000] {standard_task_runner.py:83} INFO - Job 28: Subtask sql_job
[2023-04-28T00:43:33.060+0000] {task_command.py:389} INFO - Running <TaskInstance: spark-example.sql_job scheduled__2023-04-27T00:00:00+00:00 [running]> on host 93e716072dce
[2023-04-28T00:43:33.196+0000] {taskinstance.py:1518} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=spark-example
AIRFLOW_CTX_TASK_ID=sql_job
AIRFLOW_CTX_EXECUTION_DATE=2023-04-27T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2023-04-27T00:00:00+00:00
[2023-04-28T00:43:33.466+0000] {spark_sql.py:177} INFO - /home/***/.local/lib/python3.7/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found

[2023-04-28T00:43:33.469+0000] {spark_sql.py:177} INFO - JAVA_HOME is not set

[2023-04-28T00:43:33.486+0000] {taskinstance.py:1776} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_sql.py", line 97, in execute
    self._hook.run_query()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_sql.py", line 183, in run_query
    f"Cannot execute '{self._sql}' on {self._master} (additional parameters: '{cmd}'). "
airflow.exceptions.AirflowException: Cannot execute 'select 1+1' on local (additional parameters: ''). Process exit code: 1.
[2023-04-28T00:43:33.490+0000] {taskinstance.py:1332} INFO - Marking task as FAILED. dag_id=spark-example, task_id=sql_job, execution_date=20230427T000000, start_date=20230428T004332, end_date=20230428T004333
[2023-04-28T00:43:33.507+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 28 for task sql_job (Cannot execute 'select 1+1' on local (additional parameters: ''). Process exit code: 1.; 103)
[2023-04-28T00:43:33.533+0000] {local_task_job.py:212} INFO - Task exited with return code 1
[2023-04-28T00:43:33.552+0000] {taskinstance.py:2596} INFO - 0 downstream tasks scheduled from follow-on schedule check
